start

```diff
@@ -11,43 +11,127 @@ MAX_BAZEL_MINOR=28
+}
+
 check_files_dont_exist () {
     if [ -e WORKSPACE ] || [ -e BUILD ] || [ -e BazelExample.hs ]
     then
-        echo "Current directory already has WORKSPACE and/or BUILD and/or BazelExample.hs files." >/dev/stderr
+        echo "Current directory already has WORKSPACE and/or BUILD and/or BazelExample.hs files." >&2
         exit 1
     fi
 }
 
 check_bazel_version () {
-    local actual_raw=$(bazel version | egrep '^Build label:' | egrep -o '[0-9.]+')
+    actual_raw=$(bazel version | grep -E '^Build label:' | grep -E -o '[0-9.]+')
 
-    IFS=. read actual_major actual_minor actual_patch <<EOF
+    # shellcheck disable=SC2034
+    IFS=. read -r actual_major actual_minor actual_patch <<EOF
 $actual_raw
 EOF
 
-    local expected_min=$MIN_BAZEL_MAJOR.$MIN_BAZEL_MINOR.0
-    local expected_max=$MAX_BAZEL_MAJOR.$MAX_BAZEL_MINOR.x
+    expected_min=$MIN_BAZEL_MAJOR.$MIN_BAZEL_MINOR.0
+    expected_max=$MAX_BAZEL_MAJOR.$MAX_BAZEL_MINOR.x
 
-    if [ "$actual_major" -gt "$MAX_BAZEL_MAJOR" ] || (
+    if [ "$actual_major" -gt "$MAX_BAZEL_MAJOR" ] || {
         [ "$actual_major" -eq "$MAX_BAZEL_MAJOR" ] &&
-            [ "$actual_minor" -ge "$MAX_BAZEL_MINOR" ] )
+            [ "$actual_minor" -gt "$MAX_BAZEL_MINOR" ]
+         }
     then
-	echo "Warning: a too new version of Bazel detected: v${actual_raw}."  >/dev/stderr
-	echo "         Recommended versions are from v${expected_min} to v${expected_max}."  >/dev/stderr
-    elif [ "$actual_major" -lt "$MIN_BAZEL_MAJOR" ] || (
+	      echo "Warning: a too new version of Bazel detected: v${actual_raw}."  >&2
+	      echo "         Recommended versions are from v${expected_min} to v${expected_max}."  >&2
+    elif [ "$actual_major" -lt "$MIN_BAZEL_MAJOR" ] || {
           [ "$actual_major" -eq "$MIN_BAZEL_MAJOR" ] &&
-            [ "$actual_minor" -lt "$MIN_BAZEL_MINOR" ] )
+            [ "$actual_minor" -lt "$MIN_BAZEL_MINOR" ]
+         }
     then
-        echo "Error: Need at least Bazel v${expected_min} but v${actual_raw} detected." >/dev/stderr
+        echo "Error: Need at least Bazel v${expected_min} but v${actual_raw} detected." >&2
         exit 1
     fi
 }
 
+insert_if_equal () {
+    [ "$1" = "$2" ] && printf '%s' "$3"
+}
+
+parse_args "$@"
 check_files_dont_exist
 check_bazel_version
 
-cat > WORKSPACE <<"EOF"
+bindist_toolchain=$(cat <<EOF
+# Download a GHC binary distribution from haskell.org and register it as a toolchain.
+rules_haskell_toolchains()
+EOF
+)
+
+nix_toolchain=$(cat <<EOF
+# Load nixpkgs_git_repository from rules_nixpkgs,
+# which was already initialized by rules_haskell_dependencies above.
+load(
+    "@io_tweag_rules_nixpkgs//nixpkgs:nixpkgs.bzl",
+    "nixpkgs_git_repository",
+)
+
+# Fetch a version of nixpkgs from GitHub.
+# For more information see the documentation of rules_nixpkgs at
+# https://github.com/tweag/rules_nixpkgs/blob/master/README.md
+nixpkgs_git_repository(
+    name = "nixpkgs",
+    # TODO: fix this to a git commit hash and insert sha256
```

Comment:
Having a sha256: yes. But for the revision, using a tag works just as well: tags are assigned immutably by upstream, right?

Reply:
Tags are not immutable by default, and I wouldn’t be sure they are never modified (even if by accident). Using the rev hash eliminates a source of bugs.

Reply:
Tags are good enough in this case and are better documentation that won't go out-of-date. Especially when coming from  a trusted source like the Nixpkgs release managers.

Reply:
Also, this is putting a `TODO` in every user's `WORKSPACE` file when the currently hardcoded value is a fine default.